applications/debug/unit_tests/nfc/nfc_test.c

```diff
@@ -287,9 +296,630 @@ MU_TEST(mf_classic_dict_load_test) {
     furi_record_close(RECORD_STORAGE);
 }
 
+MU_TEST(nfca_file_test) {
+    NfcDevice* nfc = nfc_device_alloc();
+    nfc->format = NfcDeviceSaveFormatUid;
+    mu_assert(nfc != NULL, "nfc_device_data != NULL assert failed\r\n");
+
+    // Fill the UID, sak, ATQA and type
+    uint8_t uid[7] = {0x04, 0x01, 0x23, 0x45, 0x67, 0x89, 0x00};
+    memcpy(nfc->dev_data.nfc_data.uid, uid, 7);
+    nfc->dev_data.nfc_data.uid_len = 7;
+
+    nfc->dev_data.nfc_data.sak = 0x08;
+    nfc->dev_data.nfc_data.atqa[0] = 0x00;
+    nfc->dev_data.nfc_data.atqa[1] = 0x04;
+    nfc->dev_data.nfc_data.type = FuriHalNfcTypeA;
+
+    // Save the NFC device data to the file
+    mu_assert(nfc_device_save(nfc, "nfca"), "nfc_device_save == true assert failed\r\n");
```

Comment:
It looks like nfc_device_save is not behaving correctly, it should take the full path to the save file as arguments.

Reply:
It was very painful, but now it does